### PR TITLE
gnrc_ipv6: replace vtimer by xtimer

### PR DESCRIPTION
**Pull request by cgundogan**
This PR replaces some vtimer calls by xtimer calls in `gnrc_ipv6`

### CODE REVIEW

File: sys/include/net/gnrc/ipv6/netif.h [hunk 1]
```diff
@@ -578,6 +586,25 @@ static inline bool gnrc_ipv6_netif_addr_is_non_unicast(const ipv6_addr_t *addr)
...
+ * @brief   Sets a xtimer.
+ *
+ * @param[in] timer     The pointer to a xtimer.
+ * @param[in] offset    Offset for the timer.
+ * @param[in] msg       The pointer to the msg_t of the timer.
+ * @param[in] msg_type  The type of the message.
+ * @param[in] msg_data  The data of the message.
+ * @param[in] pid       The pid that the timer will send a message to.
+ */
+static inline void gnrc_ipv6_set_timer(xtimer_t *timer, uint32_t offset, msg_t *msg,
```
Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: `gnrc_ipv6_netif_set_timer()`

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: But I'd rather would put it as a helper function in `xtimer` (because you also use it in `gnrc_sixlowpan`)

Comment (cgundogan) [inline sys/include/net/gnrc/ipv6/netif.h]: yes, I started without this and things looked very repetitive so I introduced this one. But I am not sure whether this should reside in xtimer. I just identified netif.h as a common header of all locations I changed

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: That's weird… `nc.h` uses also timers and they can be used independent from each other.

Comment (authmillenon) [inline sys/include/net/gnrc/ipv6/netif.h]: \+ this functionality has nothing to do with IPv6 in general, but with timers and I guess other modules using xtimer also would benefit from it.

**Comment by cgundogan** [comment]
waiting for #4117

**Comment by cgundogan** [comment]
I will slowly replace this PR with smaller PRs, because it is much simpler to maintain and review them.
First PR is here: #4121

**Comment by cgundogan** [comment]
not needed anymore

**Pull request closed without merging**